src/lib.rs

```diff
@@ -416,65 +416,59 @@ impl NeuQuant {
             self.netindex[j] = max_netpos
         } // really 256
     }
-
     /// Search for best matching color
     fn search_netindex(&self, b: u8, g: u8, r: u8, a: u8) -> usize {
-        let mut bestd = 1 << 30; // ~ 1_000_000
-        let mut best = 0;
-        // start at netindex[g] and work outwards
-        let mut i = self.netindex[g as usize];
-        let mut j = if i > 0 { i - 1 } else { 0 };
-
-        while (i < self.netsize) || (j > 0) {
-            if i < self.netsize {
-                let p = self.colormap[i];
-                let mut e = p.g - g as i32;
-                let mut dist = e * e; // inx key
-                if dist >= bestd {
-                    break;
-                } else {
-                    e = p.b - b as i32;
-                    dist += e * e;
-                    if dist < bestd {
-                        e = p.r - r as i32;
-                        dist += e * e;
-                        if dist < bestd {
-                            e = p.a - a as i32;
-                            dist += e * e;
-                            if dist < bestd {
-                                bestd = dist;
-                                best = i;
-                            }
-                        }
-                    }
-                    i += 1;
+        let mut best_dist = std::i32::MAX;
+        let first_guess = self.netindex[g as usize];
+        let mut best_pos = first_guess;
+        let mut i = best_pos;
+
+        #[inline]
+        fn sqr_dist(a: i32, b: u8) -> i32 {
+            let dist = a - b as i32;
+            dist * dist
+        }
+
+        {
+            let mut cmp = |i| {
```

Comment:
I found it a bit hard to directly see the correspondence to the previous code although it's definitely cleaner to write the search this way. Can you add comments on what the return value of this method 'means'? Or even replace it with an explicit enum:

```
enum SearchResult {
    Continue,
    Done,
}
```

Reply:
Changed it to `std::ops::ControlFlow`